File: shop/shop/src/main/java/com/onlinebookshop/shop/service/AuthorService.java
package com.onlinebookshop.shop.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.onlinebookshop.shop.model.Author;
import com.onlinebookshop.shop.repository.AuthorRepository;

@Service
public class AuthorService {
	private AuthorRepository arp;

	public AuthorService(AuthorRepository arp) {
		this.arp = arp;
	}
	
	//add new author
	public void addAuthor(Author newAuthor)
	{
		arp.save(newAuthor);		
	}
	
	//fetch all authors
	public List<Author> getAllAuthors()
	{
		return arp.listAuthors();
	}
	
	//fetch by id
	public Author getAuthById(int id)
	{
		return arp.findAuthorById(id);
	}
	
	//update authors by id
	public int updateAuthor(int id, String name)
	{
		return arp.updateAuthor(id, name);
	}
	
	//delete authors service
	public int deleteAuth(int id)
	{
		return arp.deleteAuthor(id);
	}

}

File: online-book-shop/src/services/AuthorService.js
import axios from "axios";
const API_URL="http://localhost:2020/api/authors";
class BookService {
    //fetch all books
    getAllAuthors() {
        return axios.get(API_URL);
    }
    //fetch book by id
    getAuthorkById(id){
        return axios.get(`${API_URL}/${id}`);
    }
    //add new book
    addAuthor(book){
        return axios.post(API_URL,book);
    }
    //delete a book by id
    deleteAuthor(id){
        return axios.delete(`${API_URL}/${id}`);
    }
    //update book by id
    updateAuthor(id,book){
        return axios.put(`${API_URL}/${id}`,book);
    }
}
export default new BookService();
 